### PR TITLE
decorate a class

### PR DESCRIPTION
**Pull request by BANCUNGUYE66**
### OOP school library: decorate a class

- Implement composition (as an example of the design pattern).
- Run a program using the command line.

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @BANCUNGUYE66,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### The Usage link in your menu header on the readme is not linked to any section. Kindly fix that.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super
```
Comment (Ismailco) [inline decorator.rb]: - [x] Kindly make sure to remove the `super` statement from the `Decorator` class, this produces an error 👇🏼 

Screenshot_20230620_175754

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require_relative 'decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    trimmed_name = @nameable.correct_name[0...10]
+    trimmed_name.length < @nameable.correct_name.length ? "#{trimmed_name}..." : trimmed_name
```
Comment (Ismailco) [inline trimmer_decorator.rb]: - [x] [OPTIONAL] I suggest you stick to the requirements, here you shouldn't add `...` after the trimmed name
```suggestion
    trimmed_name.length < @nameable.correct_name.length ? "#{trimmed_name}" : trimmed_name
```

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @BANCUNGUYE66, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by laiifuu — APPROVED** [review]
# STATUS: APPROVED :heavy_check_mark:

## Hi @BANCUNGUYE66  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
friends-joey

## Highlights
- [x] Required Changes implemented :heavy_check_mark: 

## Optional suggestions
- [OPTIONAL] I suggest you add a couple of future features to your README.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**